docs/mast/mast.rst

```diff
@@ -293,6 +293,74 @@ Product filtering can also be appllied directly to a table of products without p
 
                
 
+Accessing Proprietary Data
+==========================
+
+To access data that is not publicly available users may log into their
+`MyST Account <https://archive.stsci.edu/registration/index.html>`_.
+This can be done by using the `~astroquery.mast.MastClass.login` function,
+or by initializing a class instance with a username/password.
+If a password is not supplied, the user will be prompted to enter one.
+
+.. code-block:: python
+
+                >>> from astroquery.mast import Observations
+                >>> Observations.login(username="testUser@stsci.edu",password="testPwd")
+
+                Authentication successful!
+                Session Expiration: 600 minute(s)
+
+                >>> sessionInfo = Observations.session_info()
+
+                Session Expiration: 559.0 min
+                Username: testUser@stsci.edu
+                First Name: Test
+                Last Name: User
+
+              
+.. code-block:: python
+
+                >>> from astroquery.mast import Observations
+                >>> mySession = Observations(username="testUser@stsci.edu",password="testPwd")
+
+                Authentication successful!
+                Session Expiration: 600 minute(s)
+
+                >>> sessionInfo = mySession.session_info()
+
+                Session Expiration: 559.0 min
+                Username: testUser@stsci.edu
+                First Name: Test
+                Last Name: User
+
+\* For security passwords should not be typed into a terminal or Jupyter notebook
+but instead input using a more secure method such as `~getpass.getpass`.
```

Comment:
So can we just use getpass in `login`?

Reply:
Yes getpass is the default if a password isn't supplied, but if someone does want to directly supply a password that is also allowed.  I did this partly because that's what the other services with log in capabilities do, and partly because I didn't want to force user interaction for the case when someone wants to run a cron or something else automatic and would rather supply the password directly.

Reply:
OK, good plan.  I think that should be mentioned in the docs, though

Reply:
is it and I just missed it?  Sorry, I'm reviewing with 50% attention...

Reply:
In the rst documentation under Proprietary Data Access it says:
> If a password is not supplied, the user will be prompted to enter one.

And in the docstring for the log in function it says:
> If a password is necessary but not supplied it will be prompted for.

If you think I should highlight this more let me know. I want to help people not type their passwords into the terminal as much as possible!

Reply:
No, that's what I am looking for.  Sorry, too many parallel operations here.

Reply:
👍